CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+# Contributing to this repository
+
+# Table of Contents
+1. [Commits and pull requests](#commits-release-notes-pull-requests)
+    1. [Fixing a bug](#fixing-a-bug)
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests
+### Fixing a bug
+
+1. Fork the repo and commit changes to a new branch in your fork.
+2. Make pull requests against the master branch in the rpc-gating repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure:
+
+* The first line should be limited to 50 characters, should not end with a full stop and the first word should be capitalised.
+* Use the imperative mood in the subject line - e.g. ```Fix a typo in CONTRIBUTING.md```, ```Remove if/else block in myfile.sh```, ```Eat your dinner```.
+* Insert a single blank line after the first line.
+* Subsequent lines should be wrapped at 72 characters.
+* Provide a detailed description of the change in the following lines, using the guidelines in the section below.
+* The last line of the commit should be a reference to the issue being fixed using the JIRA issue Key e.g. ```RE-1500```
+
+In your commit message please consider the following points:
+
+* In the commit subject line, provide a brief description of the change.
+* The commit message must be descriptive and contain all the information required to fully understand and review the patch.
+* Do not assume the reviewer understands what the original problem was.
+* Do not assume the reviewer has access to external web services or site.
+* Do not assume the code is self-evident or self-documenting.
+* Describe why the change is being made.
+* If the commit message suggests an improved code structure, you may be able to split this into two or more commits.
+* Ensure sufficient information is provided to review your commit.
+
+
+### Pull requests
+
+* A pull request (PR) should ideally contain a single commit
```

Comment:
Why? A PR can container several commits, each targeting a specific problem to solve - whereas the PR pulls them all together into the final solution. Trying to force one commit per PR puts us in a situation where if one thing is based on another we can't continue work until the previous commit is merged. It also means it's harder to test a group of related fixes together.

Reply:
The original intent was to prevent huge scope creep in terms of commits in a PR. I can reword this to be more about making commits related to the problem being solved in the PR